File: sieve_of_eratosthenes/sieveOfEratosthenes.js
function sieveOfEratosthenes (n) {
	/*
    * Calculates prime numbers till a number n
    * :param n: Number upto which to calculate primes
    * :return: A boolean list contaning only primes
    */
    primes = new Array(n + 1);  // set all as true initially
    primes.fill(true);
    primes[0] = primes[1] = false;  //Handling case for 0 and 1
    sqrt_n = Math.ceil(Math.sqrt(n));
    for (let i = 2; i <= sqrt_n; i++)
        if (primes[i])
            for (let j = 2*i; j <= n; j+=i)
                primes[j] = false;
    return primes;
}

function main () {
	let n = 319; // number till where we wish to find primes
	primes = sieveOfEratosthenes(n);
	for (let i = 1; i <= n; i++)
        if (primes[i])
            console.log(i);
}

main();

File: sieve_of_eratosthenes/sieve_of_eratosthenes.py
import math


def sieve_of_eratosthenes(n):
    """
    Calculates prime numbers till a number n
    :param n: Number upto which to calculate primes
    :return: A boolean list contaning only primes
    """
    primes = [True] * (n + 1)  # Setting all as True initially
    primes[0] = primes[1] = False  # Since 0 and 1 are not primes
    sqrt_n = int(math.ceil(math.sqrt(n)))
    for i in range(2, sqrt_n, 1):
        if primes[i]:
            for j in range(2 * i, n + 1, i):
                primes[j] = False
    return primes


def main():
    n = 425
    primes = sieve_of_eratosthenes(n)
    for i in range(n + 1):  # Printing primes
        if primes[i]:
            print(i)


if __name__ == '__main__':
    main()
